### PR TITLE
feat(integration-service): optimised the SDK/API calls and removed re…

### LINKED ISSUE #885: [Bug]: Grock Models Not Working
### 📜 Description

Error trace:

```
agents-api-multi-tenant-1   | litellm.exceptions.BadRequestError: litellm.BadRequestError: 
OpenAIException - Error code: 400 - {'error': {'message': 'litellm.BadRequestError: GroqException - Error 
code: 400 - {\'error\': {\'message\': "\'messages.1\' : for \'role:user\' the following must
be satisfied[(\'messages.1\' : property \'tool_calls\' is unsupported, did you mean \'tool_call_id\'?)]", \'type\': \'invalid_request_error\'}}
Received Model Group=llama-3.1-70b
Available Model Group Fallbacks=None', 'type': None, 'param': None, 'code': '400'}}
```

To reproduce, create a session with `llama-3.1-70b` for example and post a message. 

### 👟 Reproduction steps

_No response_

### 👀 Have you searched previous issues to see if this has been raised before?

- [X] I checked and didn't find similar issue

### PR DESCRIPTION
**Pull request by Vedantsahai18**
…dundant files

Closes #885 

----

> [!IMPORTANT]
> Optimize SDK/API calls, improve integration handling, and update documentation in the integration service.
> 
>   - **Integration Handling**:
>     - `execute_integration.py`: Refactor to handle provider and method validation more efficiently.
>     - `browserbase.py`, `llama_parse.py`, `spider.py`: Simplify API key handling and improve retry logic.
>   - **Model and Tool Handling**:
>     - `prompt_step.py`, `chat.py`: Add hotfixes for handling specific models like Claude and Groq, and tool formatting.
>     - Remove redundant model definitions in `models.py`.
>   - **Serialization Optimization**:
>     - `codec.py`: Improve serialization/deserialization logging and handling.
>   - **Documentation Updates**:
>     - `README.md`: Update example cookbook links to reflect new paths and remove outdated references.
>   - **Miscellaneous**:
>     - `web.py`: Add GZIP middleware and optimize event loop policy for better performance.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=julep-ai%2Fjulep&utm_source=github&utm_medium=referral)<sup> for 181dc96e09ee4217c74a0214477d3975a4939f30. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: integrations-service/integrations/utils/integrations/remote_browser.py
Comment (HamadaSalhab) [inline integrations-service/integrations/utils/integrations/remote_browser.py:30]: We need to cleanup old/disconnected browsers & limit pool size

File: integrations-service/integrations/utils/integrations/spider.py
Comment (HamadaSalhab) [inline integrations-service/integrations/utils/integrations/spider.py:14]: Caching is not needed for initializing `SpiderLoader` instances as initializing an instance here doesn't require making any external connection

File: integrations-service/integrations/web.py
Comment (HamadaSalhab) [inline integrations-service/integrations/web.py:40]: Similarly, this is also not used anywhere inside this service

**Review by HamadaSalhab — CHANGES_REQUESTED** [review]

File: integrations-service/integrations/utils/http.py
Comment (Vedantsahai18) [reply integrations-service/integrations/utils/http.py:1]: Deleted the file

File: integrations-service/integrations/utils/integrations/remote_browser.py
Comment (Vedantsahai18) [reply integrations-service/integrations/utils/integrations/remote_browser.py:30]: Yet to test it out.

File: integrations-service/integrations/web.py
Comment (Vedantsahai18) [reply integrations-service/integrations/web.py:40]: Deleted it

File: agents-api/agents_api/routers/sessions/chat.py
Comment (creatorrr) [reply agents-api/agents_api/routers/sessions/chat.py:1]: Did you set session.context_overflow to “adaptive” ? Currently only None is supported

Comment (Vedantsahai18) [reply agents-api/agents_api/routers/sessions/chat.py:1]: Oops my bad!

File: integrations-service/integrations/utils/integrations/browserbase.py
Comment (creatorrr) [inline integrations-service/integrations/utils/integrations/browserbase.py:44]: Why cache these?

File: integrations-service/integrations/utils/integrations/remote_browser.py
Comment (creatorrr) [reply integrations-service/integrations/utils/integrations/remote_browser.py:38]: Is this really needed?

File: agents-api/agents_api/routers/sessions/chat.py
Comment (creatorrr) [inline agents-api/agents_api/routers/sessions/chat.py:174]: What’s this for?

File: agents-api/agents_api/activities/task_steps/prompt_step.py
Comment (creatorrr) [inline agents-api/agents_api/activities/task_steps/prompt_step.py:181]: - [ ] Add a `# FIXME: Enable formatted_tools` once format-tools PR is merged.
- [ ] Shouldn't we limit the HOTFIX only for groq if that's the only broken provider? Because this strips `tool_call_id` from **all** providers.

File: agents-api/agents_api/routers/sessions/chat.py
Comment (creatorrr) [reply agents-api/agents_api/routers/sessions/chat.py:174]: Need to do same as prompt step FIXME.

File: integrations-service/integrations/utils/integrations/remote_browser.py
Comment (creatorrr) [reply integrations-service/integrations/utils/integrations/remote_browser.py:38]: this can be problematic. if we have multiple workers, they wont be sharing the state and the browser pool, some other worker might run the browser.close() method on the same browserbase instance when hitting the max_pool_size limit

Comment (Vedantsahai18) [reply integrations-service/integrations/utils/integrations/remote_browser.py]: I have removed the browser pool limit from this PR and will create a new one

File: agents-api/agents_api/routers/sessions/chat.py
Comment (Vedantsahai18) [reply agents-api/agents_api/routers/sessions/chat.py]: Done

File: agents-api/agents_api/activities/task_steps/prompt_step.py
Comment (Vedantsahai18) [reply agents-api/agents_api/activities/task_steps/prompt_step.py]: Done

**Review by HamadaSalhab — APPROVED** [review]
LGTM! If everything is tested, feel free to merge.

**Pull request merged**
